src/ut_control_plane.c

```diff
@@ -314,27 +314,27 @@ static int callback_http(struct lws *wsi, enum lws_callback_reasons reason, void
             break;
         }
 
-        case LWS_CALLBACK_HTTP_BODY_COMPLETION: {
+        case LWS_CALLBACK_HTTP_BODY_COMPLETION:
+        {
             UT_CONTROL_PLANE_DEBUG("LWS_CALLBACK_HTTP_BODY_COMPLETION\n");
-            if(pss != NULL)
+            if (perSessionData != NULL)
             {
-                UT_CONTROL_PLANE_DEBUG("LWS_CALLBACK_HTTP_BODY_COMPLETION, pss not NULL\n");
-                msg.message = malloc((int)pss->post_data_len + 1);
+                UT_CONTROL_PLANE_DEBUG("LWS_CALLBACK_HTTP_BODY_COMPLETION, perSessionData not NULL\n");
+                msg.message = malloc((int)perSessionData->post_data_len + 1);
                 assert(msg.message != NULL);
                 if (msg.message == NULL)
                 {
                     UT_CONTROL_PLANE_ERROR("Malloc failed\n");
                     break;
                 }
-                msg.size = (int)pss->post_data_len;
+                msg.size = (int)perSessionData->post_data_len;
                 msg.status = DATA_RECIEVED;
-                memset(msg.message, 0, (int)len + 1);
-                strncpy(msg.message, (const char *)pss->post_data, pss->post_data_len);
-                msg.message[pss->post_data_len] = '\0';
-                //UT_CONTROL_PLANE_DEBUG("Received message:\n %s\n", msg.message);
+                strncpy(msg.message, (const char *)perSessionData->post_data, perSessionData->post_data_len);
+                msg.message[perSessionData->post_data_len] = '\0';
```

Comment:
U don’t need this strncpy adds one

Reply:
resolving as per the discussion we had on Teams.